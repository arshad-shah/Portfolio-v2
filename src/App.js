import React, { lazy, Suspense } from 'react';
import { ThemeProvider, createTheme, responsiveFontSizes } from '@material-ui/core/styles';
import { useMediaQuery } from '@material-ui/core';
import CssBaseline from '@material-ui/core/CssBaseline';
import Fab from '@material-ui/core/Fab';
import ArrowUpwardRoundedIcon from '@material-ui/icons/ArrowUpwardRounded';
import ScrollToTop, { ScrollToFade } from './App.util';
import useStyles from './styles/App.styles';
import Loading from './components/Loading';
import Mobile from './components/navigation/Mobile';

const Header = lazy(() => import('./components/Header'));
const LandingPage = lazy(() => import('./pages/LandingPage'));
const Aboutme = lazy(() => import('./pages/Aboutme'));
const Projects = lazy(() => import('./pages/Projects'));
const Footer = lazy(() => import('./components/Footer'));
let theme = createTheme({
	palette: {
		type: 'light',
		primary: {
			main: '#fdff95',
		},
		secondary: {
			main: '#ffbf0d',
		},
		action: {
			hoverOpacity: 0.09,
		},
	},
	typography: {
		fontFamily: 'Nunito',
		fontSize: 18,
	},
});

theme = responsiveFontSizes(theme);

const renderLoader = () => <Loading />;

export default function App(props) {
	const classes = useStyles();
	const isMobile = useMediaQuery(theme.breakpoints.down('sm'));

	return (
		<ThemeProvider theme={theme}>
			<div className={classes.root}>
				<Suspense fallback={renderLoader()}>
					<CssBaseline />
					<Header isMobile={isMobile} data-testid="header" />

					{isMobile ? <Mobile /> : null}

					<ScrollToFade {...props}>
						<LandingPage isMobile={isMobile} />
					</ScrollToFade>

					<Aboutme />

					<Projects data-testid="projectsSection" />

					<ScrollToTop {...props}>
						<Fab
							color="secondary"
							data-block="backtotopbutton"
							data-testid="backtoTopButton"
							size="large"
							aria-label="scroll back to top">
							<ArrowUpwardRoundedIcon fontSize="large" />
						</Fab>
					</ScrollToTop>

					<Footer isMobile={isMobile} />
				</Suspense>
			</div>
		</ThemeProvider>
	);
}
