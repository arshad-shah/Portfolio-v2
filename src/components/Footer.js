import { Box, IconButton, Typography } from "@material-ui/core";
import React from "react";
import useStyles from "./Footer.styles";

// icons
import GitHubIcon from "@material-ui/icons/GitHub";
import LinkedInIcon from "@material-ui/icons/LinkedIn";
import MailIcon from "@material-ui/icons/Mail";

export default function Footer() {
  const classes = useStyles();
  return (
    <Box className={classes.root}>
      <Typography variant="h6">
        &copy; {new Date().getFullYear()} Designed and Created By: Arshad Shah
      </Typography>
      <Box>
        <IconButton
          role="button"
          className={classes.socialLinks}
          color="default"
          aria-label="link to Github account of Arshad shah"
          data-block="link to Github"
          href="https://github.com/arshad-shah"
        >
          <GitHubIcon />
        </IconButton>

        <IconButton
          role="button"
          className={classes.socialLinks}
          color="default"
          aria-label="link to Linkedin account of Arshad shah"
          data-block="link to Linkedin"
          href="https://www.linkedin.com/in/arshadshah"
        >
          <LinkedInIcon />
        </IconButton>

        <IconButton
          className={classes.socialLinks}
          color="default"
          role="button"
          aria-label="link to mail account of Arshad shah"
          data-block="link to Mail"
          href="mailto:shaharshad1999@outlook.com"
        >
          <MailIcon />
        </IconButton>
      </Box>
      <Typography variant="subtitle2">
        Google Play and the Google Play logo are trademarks of Google LLC.
      </Typography>
      <Typography variant="subtitle2">V2.0 Date: 15/August/2021</Typography>
    </Box>
  );
}
