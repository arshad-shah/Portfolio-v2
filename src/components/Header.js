import React, { useState } from "react";
import { AppBar, Toolbar, Typography } from "@material-ui/core";
import HideOnScroll from "./HideOnScroll";

import Desktop from "./navigation/Desktop";
import Mobile from "./navigation/Mobile";
import useStyles from "../styles/Header.styles";
import logo from "../assets/logo.svg";
import PropTypes from "prop-types";

export default function Header(props) {
  const classes = useStyles();
  const { isMobile } = props;

  const [click, setClick] = useState(false);
  const handleClick = () => setClick(!click);
  const closeMobileMenu = () => {
    setClick(false);
  };
  const handleMenu = () => {
    closeMobileMenu();
    handleClick();
  };

  return (
    <div className={classes.root}>
      <Toolbar id="back-to-top-anchor" />
      <HideOnScroll {...props}>
        <AppBar data-testid="header">
          <Toolbar>
            <img aria-label="Logo" src={logo} className={classes.logo} alt="" />
            <Typography
              variant="h5"
              component="p"
              color="textPrimary"
              className={classes.title}
            >
              A SHAH
            </Typography>
            {isMobile ? (
              <Mobile
                click={click}
                closeMobileMenu={closeMobileMenu}
                handleMenu={handleMenu}
              />
            ) : (
              <Desktop />
            )}
          </Toolbar>
        </AppBar>
      </HideOnScroll>
    </div>
  );
}
Header.propTypes = {
  isMobile: PropTypes.bool,
};
