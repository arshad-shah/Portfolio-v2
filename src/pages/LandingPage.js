import React from 'react';
import { Box, Typography, Button, IconButton, Tooltip } from '@material-ui/core';
import { Link } from 'react-scroll';

import ArrowDownwardRoundedIcon from '@material-ui/icons/ArrowDownwardRounded';
// icons
import GitHubIcon from '@material-ui/icons/GitHub';
import LinkedInIcon from '@material-ui/icons/LinkedIn';
import MailIcon from '@material-ui/icons/Mail';
import useStyles from '../styles/LandingPage.styles';

import PropTypes from 'prop-types';

export default function LandingPage({ isMobile }) {
	const classes = useStyles();
	return (
		<Box className={classes.hero} data-testid="hero">
			<Typography className={classes.landingIntro} variant="h4">
				Hi, My Name Is -
			</Typography>

			<Typography className={classes.landingName} variant="h1">
				Arshad Shah!
			</Typography>

			<Typography className={classes.landingDesc} variant="h4">
				A Third Year <b>Computer Science</b> Student and a <b>Software Engineer</b>.
			</Typography>

			<Box className={classes.Links}>
				<Tooltip title="Github Profile" arrow>
					<IconButton
						data-testid="githubForHero"
						role="button"
						className={classes.socialLinks}
						color="primary"
						aria-label="link to Github account of Arshad shah"
						data-block="linktoGithub"
						href="https://github.com/arshad-shah">
						<GitHubIcon fontSize="large" />
					</IconButton>
				</Tooltip>

				<Tooltip title="LinkedIn Profile" arrow>
					<IconButton
						data-testid="linkedInForHero"
						role="button"
						className={classes.socialLinks}
						color="primary"
						aria-label="link to Linkedin account of Arshad shah"
						data-block="linktoLinkedin"
						href="https://www.linkedin.com/in/arshadshah">
						<LinkedInIcon fontSize="large" />
					</IconButton>
				</Tooltip>

				<Tooltip title="Email Me" arrow>
					<IconButton
						data-testid="mailForHero"
						className={classes.socialLinks}
						color="primary"
						role="button"
						aria-label="link to mail account of Arshad shah"
						data-block="linktoMail"
						href="mailto:arshad@arshadshah.com">
						<MailIcon fontSize="large" />
					</IconButton>
				</Tooltip>
			</Box>
			{!isMobile ? (
				<Tooltip title="Scroll Down" arrow>
					<Link to="aboutme" spy smooth>
						<Button
							data-testid="scrollDown"
							aria-label="scroll down button"
							size="large"
							color="secondary"
							variant="outlined"
							data-block="scrolldown">
							<ArrowDownwardRoundedIcon fontSize="large" />
						</Button>
					</Link>
				</Tooltip>
			) : null}
		</Box>
	);
}

LandingPage.propTypes = {
	isMobile: PropTypes.bool.isRequired,
};
