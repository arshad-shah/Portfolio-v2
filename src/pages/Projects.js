import { Box, Grid, Toolbar, Typography } from "@material-ui/core";
import React from "react";
import useStyles from "./styles/Projects.styles";
import ProjectCard from "../components/ProjectCard";
import { cardData } from "../data/projectData";

export default function Projects() {
  const classes = useStyles();
  return (
    <Box className={classes.root}>
      <Toolbar className={classes.projectsLink} id="projects" />
      <Typography align="center" gutterBottom variant="h2">
        Projects
      </Typography>
      <Grid
        container
        spacing={4}
        justifyContent="space-around"
        alignContent="center"
        alignItems="center"
      >
        {cardData.map((data, key) => (
          <Grid key={key} aria-label="ProjectCard" item lg={4} xs={12} sm={6}>
            <ProjectCard
              heading={data.heading}
              description={data.description}
              languages={data.languages}
              linkAriaLabel={data.linkAriaLabel}
              repoLink={data.repoLink}
              siteLink={data.siteLink}
              shopLink={data.shopLink}
            />
          </Grid>
        ))}
      </Grid>
    </Box>
  );
}
