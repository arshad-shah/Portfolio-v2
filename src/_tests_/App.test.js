import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import Header from '../components/Header';
import Footer from '../components/Footer';
import LandingPage from '../pages/LandingPage';
import AboutMe from '../pages/AboutMe';
import Projects from '../pages/Projects';

describe('the full app:', () => {
	it('should render the full app', () => {
		const { container } = render(<App />);
		expect(container).toMatchSnapshot();
	});

	it('should render the Header', () => {
		const { container } = render(<Header />);
		expect(container).toBeInTheDocument();
	});

	it('should render the Landing page', () => {
		const { container } = render(<LandingPage />);
		expect(container).toBeInTheDocument();
	});

	it('should render the About', () => {
		const { container } = render(<AboutMe />);
		expect(container).toBeInTheDocument();
	});

	it('should render the projects', () => {
		const { container } = render(<Projects />);
		expect(container).toBeInTheDocument();
	});

	it('should render the Footer', () => {
		const { container } = render(<Footer />);
		expect(container).toBeInTheDocument();
	});
});
