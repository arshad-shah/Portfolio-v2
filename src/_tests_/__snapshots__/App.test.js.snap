// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the full app: should render the full app 1`] = `
<div>
  <div
    class="App-root-1"
  >
    <div
      style="opacity: 1; display: none; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      class="makeStyles-root-3"
      data-block="backtotopbutton"
      role="presentation"
      style="transform: scale(0); visibility: hidden; display: none;"
    >
      <button
        aria-label="scroll back to top"
        class="MuiButtonBase-root MuiFab-root App-fab-2 MuiFab-secondary"
        data-block="backtotopbutton"
        data-testid="backtoTopButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="19"
              y2="5"
            />
            <polyline
              points="5 12 12 5 19 12"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="Loading-root-5"
    >
      <div
        class="MuiCircularProgress-root Loading-progress-7 MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 100px; height: 100px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="19.5"
            stroke-width="5"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root Loading-name-6 MuiTypography-h3"
      >
        Loading...
      </h3>
    </div>
  </div>
</div>
`;
